[4924] Switch 3.1.x sites to run within Next UI

rart opened the pull request:
craftercms/craftercms#4924

rart committed:
[4924] Switch 3.1.x sites to run within Next UI
craftercms/craftercms#4924

rart committed:
Keep boolean value regardless of argument type

rart committed:
Remove traces of previewChoice setting, consolidate preview url and preview routing on a common constant/util

rart committed:
Remove fixed readonly flag opening form for legacy pencils

sumerjabri commented on static-assets/components/cstudio-common/common-api.js:
```diff
@@ -1585,11 +1585,12 @@ var nodeOpen = false,
-          } else {
-            window.location = `/studio/preview#/?page=${url}&site=${CStudioAuthoringContext.site}`;
-          }
+          window.location = CrafterCMSNext.util.system.getSystemLink({
```
Upon site load/open, read site-config.xml and find the property:
```
<preview-compatibilty>craftercmsv3</>
```

If the value is `craftercmsv3`, then use the old UI, otherwise use the new UI. If the property is missing or not equal to that value, then use the new UI.

sumerjabri commented on static-assets/scripts/main.js:
```diff
@@ -531,10 +531,12 @@
-            $window.location.href = `${
-              previewChoice && previewChoice[site.id] === '2' ? '/studio/next/preview' : '/studio/preview'
-            }#/?page=/&site=${site.id}`;
+            $window.location.href = CrafterCMSNext.util.system.getSystemLink({
```
Delete this file?

sumerjabri commented on ui/app/src/components/SiteSwitcherSelect/SiteSwitcherSelect.tsx:
```diff
@@ -37,13 +37,12 @@ function SiteSwitcherSelect(props: SiteSwitcherSelectProps) {
   const onSiteChange = ({ target: { value } }) => {
     if (!isBlank(value) && site !== value) {
-      if (previewChoice[value] === '2') {
+      if (window.location.href.includes(PREVIEW_BASE_URL)) {
```
Switch to using the `site-config.xml` preference.

sumerjabri commented on ui/app/src/components/SiteSwitcherSelect/SiteSwitcherSelect.tsx:
```diff
@@ -52,7 +51,6 @@ function SiteSwitcherSelect(props: SiteSwitcherSelectProps) {
             (window.location.href = getSystemLink({
               site: value,
               systemLinkId: 'preview',
-              previewChoice,
```
This is what we probably keep.

rart committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4924

 Conflicts:
	ui/app/src/components/PathNavigator/PathNavigator.tsx

rart committed:
[4924] Switch 3.1.x sites to run within Next UI
Incorporate optional site config preference for using legacy preview
Dead code removal

rart committed:
Add TODO note for future backend changes

rart committed:
Adjust config reloading logic for ui config and site-config
